matrix-notifications.el

```diff
@@ -25,39 +25,112 @@
 
 ;;;; Customization
 
-(defcustom matrix-client-enable-notifications t
-  "Enable notifications for incoming messages."
-  :type 'boolean
+(defgroup matrix-client-notifications nil
+  "Notification settings."
   :group 'matrix-client)
 
+(defcustom matrix-client-enable-notifications t
+  "Enable notifications for incoming messages in uncustomized rooms.
+Notifications for individual rooms can be overridden in
+`matrix-client-notifications-rooms'."
+  :type 'boolean)
+
+(defcustom matrix-client-notify-dispatcher #'matrix-client-notify-dispatch
```

Comment:
Thanks for this, I actually prefer [configuring notifications with a function](https://github.com/jgkamat/dotfiles/blob/master/emacs/.emacs.d/lisp/circe-config.el#L252-L272), and I wanted this eventually, but you beat me to it :)